perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -143,11 +143,14 @@ public PerunSession getPerunSession(PerunPrincipal principal, PerunClient client
 
 				if (client.getType() != PerunClient.Type.OAUTH) {
 					// Try to update LoA for userExtSource
-					UserExtSource ues;
+					UserExtSource ues = null;
 					String shibIdentityProvider = principal.getAdditionalInformations().get(UsersManagerBl.ORIGIN_IDENTITY_PROVIDER_KEY);
+					boolean findByExtSourceNameAndLogin = true;
 					if(shibIdentityProvider != null && extSourcesWithMultipleIdentifiers.contains(shibIdentityProvider)) {
 						ues = usersManagerBl.getUserExtSourceFromMultipleIdentifiers(internalSession, principal);
```

Comment:
We should probably catch `UserExtSourceNotExistsException` here so we can try to find ues by name and login in such a case (similarly to `getUserByExtSourceInformation` method). What do you think?

Reply:
You are right. I updated this block of code.